### PR TITLE
ref(ts) Convert actionCreator/tags to typescript

### PR DESCRIPTION
**Pull request by markstory**
I had to rejig a few types in globalSelectionHeader because the getParams() function was unable to accept a GlobalSelection header value which is used in a few places.

### CODE REVIEW

File: src/sentry/static/sentry/app/actionCreators/tags.tsx [hunk 1]
```diff
@@ -47,9 +51,10 @@ const BUILTIN_TAGS = [
   'stack.stack_level',
 ].map(tag => ({
   key: tag,
+  name: tag,
```
Comment (markstory) [inline src/sentry/static/sentry/app/actionCreators/tags.tsx:54]: Needed to satisfy the Tag type.

File: src/sentry/static/sentry/app/components/charts/utils.tsx [hunk 2]
```diff
@@ -20,11 +21,7 @@ export const AREA_COLORS = [
   {line: '#422C6F', area: '#422C6F'},
 ];
 
-export type DateTimeObject = {
-  start: Date | null;
-  end: Date | null;
-  period?: string;
-};
+export type DateTimeObject = Partial<GlobalSelection['datetime']>;
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/charts/utils.tsx:24]: We have a few definitions of a DateTimeObject in the app, and they are all slightly different. I thought that normalizing on the GlobalSelection definition would be an improvement as that is generally what the values should be.

File: src/sentry/static/sentry/app/components/organizations/globalSelectionHeader/getParams.tsx [hunk 3]
```diff
@@ -66,7 +66,7 @@ function getStatsPeriodValue(
 // This format was transformed to the form that moment.js understands using
 // https://gist.github.com/asafge/0b13c5066d06ae9a4446
 const normalizeDateTimeString = (
-  input: string | undefined | null
+  input: Date | string | undefined | null
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/organizations/globalSelectionHeader/getParams.tsx:69]: Moment is used in this function and also accepts Date.

File: src/sentry/static/sentry/app/components/organizations/globalSelectionHeader/getParams.tsx [hunk 4]
```diff
@@ -113,32 +115,32 @@ const getUtcValue = (maybe: string | string[] | undefined | null): string | unde
...
   statsPeriod?: ParamValue;
-  utc?: ParamValue;
+  utc?: boolean | ParamValue;
+  [others: string]: any;
 };
-type RestParams = {[others: string]: ParamValue};
 
 export function getParams(
-  params: Params & RestParams,
+  params: InputParams,
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/organizations/globalSelectionHeader/getParams.tsx:141]: This was widened to allow Date instances to be passed in. The indexed type had to be widened as well unfortunately.

**Review by scefali — APPROVED** [review]

**Review by billyvg — APPROVED** [review]

**Pull request merged**